initialization/initialize.go

```diff
@@ -203,50 +203,42 @@ func (init *Initializer) initFile(id []byte, fileIndex int, labelGroupsPerFile u
 	errChan := make(chan error, 0)
 	finishedChan := make(chan struct{}, 0)
 	batchSize := 100
-	chanBuffer := 100
+	chanBufferSize := 100
 
 	// CPU workers.
 	fileOffset := uint64(fileIndex) * labelGroupsPerFile
 	for i := 0; i < numOfWorkers; i++ {
 		i := i
-		workersOutputChans[i] = make(chan [][]byte, chanBuffer)
+		workersOutputChans[i] = make(chan [][]byte, chanBufferSize)
 		workerOffset := i
 		go func() {
 			// Calculate labels in groups and write them to channel in batches.
-			iterator := 0
-			position := uint64(workerOffset)
+			position := uint64(workerOffset * batchSize)
 			batch := make([][]byte, batchSize)
 			for position < labelGroupsPerFile {
-				batch[iterator%batchSize] = CalcLabelGroup(id, position+fileOffset, Difficulty(init.cfg.Difficulty))
+				batch[position%uint64(batchSize)] = CalcLabelGroup(id, position+fileOffset, Difficulty(init.cfg.Difficulty))
 
-				if iterator%batchSize == batchSize-1 {
+				if position%uint64(batchSize) == uint64(batchSize-1) {
 					workersOutputChans[i] <- batch
 					batch = make([][]byte, batchSize)
+					position += uint64(numOfWorkers * batchSize)
```

Comment:
you're adding to much, you should add `uint64((numOfWorkers-1)*batchSize)`. How come there is no UT that failed on this? As I understand the code is broken this way.... Please add UT and make sure that it fails and then fix it

Reply:
Good catch. The necessary UT fix wasn't on this branch. Will add it now.